sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/jwt/RestOperationsResourceRetriever.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.aad.implementation.jwt;
+
+import com.nimbusds.jose.util.Resource;
+import com.nimbusds.jose.util.ResourceRetriever;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.Assert;
+import org.springframework.web.client.RestOperations;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Arrays;
+
+/**
+ * A ResourceRetriever that accepts an RestOperations as a constructor parameter.
+ * Copied from org.springframework.security.oauth2.jwt.NimbusJwtDecoder.JwkSetUriJwtDecoderBuilder.RestOperationsResourceRetriever
+ */
+public class RestOperationsResourceRetriever implements ResourceRetriever {
+
+    private static final MediaType APPLICATION_JWK_SET_JSON = new MediaType("application", "jwk-set+json");
+
+    private final RestOperations restOperations;
+
+    public RestOperationsResourceRetriever(RestOperations restOperations) {
+        Assert.notNull(restOperations, "restOperations cannot be null");
+        this.restOperations = restOperations;
+    }
+
+    @Override
+    public Resource retrieveResource(URL url) throws IOException {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON, APPLICATION_JWK_SET_JSON));
+        ResponseEntity<String> response = getResponse(url, headers);
+        if (response.getStatusCodeValue() != HttpStatus.OK.value()) {
+            throw new IOException(response.toString());
+        }
+        return new Resource(response.getBody(), "UTF-8");
```

Comment:
The second parameter should be the content type, not the charset. In the `com.nimbusds.jose.util.DefaultResourceRetriever#retrieveResource`, it will change the exception `ClassCastException`  to `ClassCastException`, I think this should be added in current method.

Reply:
1. About `contentType`
1.1. The `Resource`'s java doc didn't say what it should be like. 
![image](https://user-images.githubusercontent.com/13167207/187103687-bf528509-b61b-440a-b1db-627a418ed928.png)
1.2. Current callers also set the content type to `UTF-8`
![image](https://user-images.githubusercontent.com/13167207/187103770-91b4a8a1-b828-43d0-9654-1a4014c85ff7.png)
1.3. This class is copied from spring-security framework. Let's keep it same with upstream. If there is some but, we should let spring-security framework update it first.

2. About `ClassCastException `. 
Same to `1.3`. I will not change this.